### PR TITLE
My Site Dashboard: Extract NewBlogDetailHeaderView to SitePickerViewController

### PR DESCRIPTION
**Pull request by momo-ozawa**
Part of #17740

## Description
This PR hides the `NewBlogDetailHeaderView` in the table view belonging to `BlogDetailsViewController` and adds a `NewBlogDetailHeaderView` to`SitePickerViewController`.
- Embeds a `SitePickerViewController`, which contains a  `NewBlogDetailHeaderView` to the top of the stack view in `MySiteViewController`
- Hides the QuickActions row in the `NewBlogDetailHeaderView` displayed in `MySiteViewController`
- `SitePickerViewController` conforms to `BlogDetailHeaderViewDelegate`

Before | After
-- | --
 | 

## Notes
- These changes are behind a feature flag.
- There may be other logic that needs to be copied over from BlogDetailsVC in the future. For this PR, I've copied over only the logic needed to implement the `BlogDetailHeaderViewDelegate` to `SitePickerViewController`.

## How to review

To make sure there's no regressions in the `BlogDetailHeaderViewDelegate`, please review the implementation commits one by one by comparing it to the [implementation](https://github.com/wordpress-mobile/WordPress-iOS/blob/a08cc72e4bfb15d4fa88c06c549c16941fb2a64a/WordPress/Classes/ViewRelated/Blog/Blog%20Details/BlogDetailsViewController.m#L1097-L1165) in `BlogDetailsViewController`.

- ae1f72c68e3041aad0aebd72bd4a214c32db5a18 (siteSwitcherTapped)
- 6cc75d8254780f83a338025e62db0ea2f40a41af (siteTitleTapped)
- a29189b404258ba76e21f2d6176e1da798f6e934 (siteIconShouldAllowDroppedImages)
- cfec913f24269045c6f01d0860acdf874aa80984 (siteIconTapped)
- 10e0b1d49f65b3b357ac6a640b533cb1312d26ca (siteIconReceivedDroppedImage)
- ef5f233adeedbdf671a68a38bc859935231fcbf6 (visitSiteTapped)

## How to test

Make sure the My Site Dashboard feature flag is **enabled**.

**siteSwitcherTapped**
1. Go to My Site
2. Tap on the site switcher
3. Switch to a different site
4. ✅ Notice that the site switcher displays the selected site
5. ✅ Notice that the site menu (BlogDetailsVC) has updated to the selected blog

**siteTitleTapped**
1. Go to My Site
2. Tap on the site title
3. Change your site title
4. Tap save
5. ✅ Notice that your site title has been updated

**siteIconTapped**
1. Go to My Site
2. Tap on the site icon
3. Update your site icon
4. ✅ Notice that your site icon has been updated
5. Remove your site icon
6. ✅ Notice that your site icon has been removed

**visitSiteTapped**
1. Go to My Site
2. Tap on the site url in the header view
3. ✅ Notice that your site is displayed in a web vc

## Regression Notes
1. Potential unintended areas of impact
BlogDetailsHeaderView regression

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I tested the above flows manually

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by momo-ozawa** [comment]
@leandroalonso Ready for review -- extracted the `NewBlogDetailHeaderView` to `SitePickerViewController`, which is then embedded as a child vc in `MySiteViewController`

**Comment by pachlava** [comment]
Hey @momo-ozawa, thanks for providing the detailed tests with the PR! I did some testing on my side, and so far I have only one really minor thing. Maybe it's planned for rework anyway. There's an empty space at the top of the Site Menu, which looked OK with the old layout, but after the rearrangement it becomes more obvious, especially under the dark theme:

photo_2022-01-20 12 14 30

Here's with the light theme:

photo_2022-01-20 12 08 27

It looks like according to designs it should not be there. However, when removed, the whole screen will look a bit off under light theme, as long as the background for the segmented control remains white (it looks like it should be grey, from designs, as well as the background for the site title).

**Comment by momo-ozawa** [comment]
@pachlava 👋 

Thanks for your feedback! This PR is more of an architectural PR, and design refinements will be addressed later once the foundational PR's have landed 🙏

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift [hunk 1]
```diff
@@ -105,6 +107,24 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
         NotificationCenter.default.addObserver(self, selector: #selector(showAddSelfHostedSite), name: .addSelfHosted, object: nil)
     }
 
+    private func showSitePicker(for blog: Blog) {
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:111]: There's an issue with displaying the site picker for the following flow:

1. Turn the My Site Dashboard flag to on
2. Log out
3. Login again and select a site
4. Tap "Show me around"
5. No site picker is shown

File: WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+QuickStart.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import Foundation
+
+private var alertWorkItem: DispatchWorkItem?
+private var observer: NSObjectProtocol?
+
+extension SitePickerViewController {
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+QuickStart.swift:6]: I tested QuickStart with this new `SitePickerViewController` and the glowing signal is not appearing at all. This is how it is with the flag disabled:

Screen Shot 2022-01-20 at 12 35 03

I'm still able to complete the quick start steps, though.

**Review by leandroalonso — COMMENTED** [review]
@momo-ozawa I tested different flows and compared the current behavior with the current `trunk`.

There were a few issues that I found that are shared in this review. I also double-checked the events to make sure we're not removing any of them by mistake.

**Comment by leandroalonso** [comment]
@momo-ozawa another thing that I noticed is that pull to refresh doesn't update site details anymore (made on another device).

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:111]: Fixed! 7884be2

File: WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+QuickStart.swift (hunk 2 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController+QuickStart.swift:6]: Fixed! 72d2296

**Comment by momo-ozawa** [comment]
@leandroalonso  👋 

> @momo-ozawa another thing that I noticed is that pull to refresh doesn't update site details anymore (made on another device).

I'll address this in a different PR, when I work on [this issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/17784) 

I've addressed the other issues you found -- ready for a re-review! 🙇‍♀️ 

https://user-images.githubusercontent.com/6711616/150538469-7921ff0e-5b5d-425a-af7a-75be3ee86566.mp4

**Review by leandroalonso — APPROVED** [review]
@momo-ozawa tested all the flows, all good! :shipit:

**Pull request merged**